Feat:  Updated the "Download" option to be a push row in both the bulk actions menu and in the More secondary action menu

waterim opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->


### Fixed Issues
<!---
1. Please postfix `$` with a URL link to the GitHub issue this Pull Request is fixing. For example, `$ https://github.com/Expensify/App/issues/<issueID>`.
2. Please postfix  `PROPOSAL:` with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).  For example, `PROPOSAL: https://github.com/Expensify/App/issues/<issueID>#issuecomment-1369752925`

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking and its automation will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<issueID>
$ https://github.com/Expensify/App/issues/<issueID(comment)>

Do NOT only link the issue number like this: $ #<issueID>
--->
$ https://github.com/Expensify/App/issues/58316
PROPOSAL: N/A


### Tests

Test 1: 
1. Go to expense report
2. Click "More" button
3. Observe new Export option
4. Go inside and observe

Test 2: 
(to update)


- [x] Verify that no errors appear in the JS console

### Offline tests
<!---
Add any relevant steps that validate your changes work as expected in a variety of network states e.g. "offline", "spotty connection", "slow internet", etc. Manual test steps should be written so that your reviewer and QA testers can repeat and verify one or more expected outcomes. If you are unsure how the behavior should work ask for advice in the `#expensify-open-source` Slack channel.
--->

Same as tests

### QA Steps
Same as tests

- [X] Verify that no errors appear in the JS console

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [X] I linked the correct issue in the `### Fixed Issues` section above
- [X] I wrote clear testing steps that cover the changes made in this PR
    - [X] I added steps for local testing in the `Tests` section
    - [X] I added steps for the expected offline behavior in the `Offline steps` section
    - [X] I added steps for Staging and/or Production testing in the `QA steps` section
    - [X] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [X] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I ran the tests on **all platforms** & verified they passed on:
    - [X] Android / native
    - [X] Android / Chrome
    - [X] iOS / native
    - [x] iOS / Safari
    - [X] MacOS / Chrome / Safari
    - [X] MacOS / Desktop
- [X] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [X] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [X] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] I verified that if a function's arguments changed that all usages have also been updated correctly
- [X] If a new component is created I verified that:
    - [X] A similar component doesn't exist in the codebase
    - [X] All props are defined accurately and each prop has a `/** comment above it */`
    - [X] The file is named correctly
    - [X] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [X] The only data being stored in the state is data necessary for rendering and nothing else
    - [X] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [X] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [X] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [X] All JSX used for rendering exists in the render method
    - [X] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [X] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos

<details>


<img width="359" alt="image" src="https://github.com/user-attachments/assets/44d0c7b0-b2e1-4142-af70-b043f74b8729" />

<img width="351" alt="image" src="https://github.com/user-attachments/assets/93ea3393-9714-45c9-9ae2-63721209f50c" />

<img width="352" alt="image" src="https://github.com/user-attachments/assets/192b0a0d-f57f-4213-88a1-94e9b3960bdd" />




</details>

waterim committed:
update more button with a new export

waterim committed:
fix tests, fix linter

waterim committed:
fix computed Options

waterim commented:
@flaviadefaria Opened a PR for initial review to understand if Im on the right path and when I get your answer on [this](https://github.com/Expensify/App/issues/58316#issuecomment-2895122080) I will update that place also with the same behaviour

waterim committed:
Merge remote-tracking branch 'upstream/main' into feature/download-option

luacmartins commented:
@waterim is this PR ready for review?

waterim committed:
Merge remote-tracking branch 'upstream/main' into feature/download-option

waterim committed:
remove artifacts after a merge conflict

waterim committed:
fix tests conflicts

waterim committed:
fix typo

flaviadefaria commented:
@dubielzyk-expensify can you help point @waterim to the correct pattern? 

@waterim should we trigger an app build here to make sure that all is working as intended?

waterim committed:
Merge remote-tracking branch 'upstream/main' into feature/download-option

dubielzyk-expensify commented:
When you click Export, this PR is using the wrong Export row. This is a comparison:

![CleanShot 2025-05-28 at 11 09 52@2x](https://github.com/user-attachments/assets/918972ff-0753-44cb-a894-22259f20eb04)

We use this correct pattern in other parts of the app already.

flaviadefaria commented:
Thanks for clarifying @dubielzyk-expensify ❤️

waterim committed:
Merge remote-tracking branch 'upstream/main' into feature/download-option

waterim committed:
fix tests

waterim committed:
Merge remote-tracking branch 'upstream/main' into feature/download-option

waterim committed:
Merge remote-tracking branch 'upstream/main' into feature/download-option

waterim committed:
Merge remote-tracking branch 'upstream/main' into feature/download-option

waterim committed:
fix UI

waterim committed:
update hook for multiple transactions

dubielzyk-expensify commented:
Can we re-upload screenshots for all platforms?

dubielzyk-expensify commented:
Pulled the branch locally and it looks good to me:

![CleanShot 2025-06-03 at 11 20 28@2x](https://github.com/user-attachments/assets/00cbfbe3-c29d-4387-9b59-dafd05dc43bf)

waterim commented:
Looks like you are looking in the search tab, Ive added it on the selected multiple transactions on the report page

flaviadefaria commented:
>Looks like you are looking in the search tab, Ive added it on the selected multiple transactions on the report page

Yeah, I think there was a miscommunication here. The search tab is now called "Reports" tab. I did see that you added it on the selected multiple transactions on the report page but we also need to do the same thing in the Reports/Search tab.

waterim committed:
Merge remote-tracking branch 'upstream/main' into feature/download-option

waterim committed:
update also search

waterim commented:
@flaviadefaria updated everything

waterim committed:
Merge remote-tracking branch 'upstream/main' into feature/download-option

waterim committed:
fix spell check

flaviadefaria commented:
>@flaviadefaria updated everything and resolved conflicts

Woo amazing! I'll trigger another test app build.

waterim committed:
Merge remote-tracking branch 'upstream/main' into feature/download-option

flaviadefaria commented:
@waterim I just tested the PR, and it looks like we're almost there! I can see the changes were made on the Reports/Search tab (thank you) however, I found two issues:

1. When exporting a full report from the Inbox we display the integration options correctly, however `Mark as manually exported` should actually just be `Mark as exported` and it should also have the integration icon next to it as is refers to marking a report as already being exported to the accounting system. We did consider the checkmark icon but eventually decided against it (as you can see in the final mockups).

<img width="1494" alt="image" src="https://github.com/user-attachments/assets/5b998536-4f53-4322-9b48-16a6d0413ef9" />

2. Then, in the Reports/Search tab, when I select a full report, I do not see the options to export to QBO and mark as exported to QBO:

<img width="1510" alt="image" src="https://github.com/user-attachments/assets/dad9fcea-9e85-4448-9fe1-a0aa02647956" />

Let me know in case you have any questions!

luacmartins commented on src/components/MoneyReportHeader.tsx:
```diff
@@ -510,6 +510,64 @@ function MoneyReportHeader({
+                },
+            },
+            [CONST.REPORT.EXPORT_OPTIONS.EXPORT_TO_INTEGRATION]: {
+                // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```
Why is this disable needed?

waterim replied:
actually no need now, just was moving it and this disable was already in the code, good catch, thanks

waterim replied:
That was already there, but that was needed for this line: `connectionName: connectedIntegration!`

luacmartins reviewed (commented):
Nice. Code-wise this looks good to me. Just pending on addressing @flaviadefaria's comments

waterim committed:
Merge remote-tracking branch 'upstream/main' into feature/download-option

waterim commented:
@flaviadefaria 

> Then, in the Reports/Search tab, when I select a full report, I do not see the options to export to QBO and mark as exported to QBO:

We didn't have this functionality at all there, should I migrate to search tab also?

waterim committed:
update icon and translation

flaviadefaria commented:
@waterim good question - let me double check this in Slack to be 100% sure.

flaviadefaria commented:
@waterim I checked internally (you're cc'ed on the thread) and @JmillsExpensify confirmed that the accounting integration export options will be implemented in this other [GH](https://github.com/Expensify/Expensify/issues/505933) along with some additional functionality. So for this PR, all that needs to be fixed is this:

>When exporting a full report from the Inbox we display the integration options correctly, however Mark as manually exported should actually just be Mark as exported and it should also have the integration icon next to it as is refers to marking a report as already being exported to the accounting system. We did consider the checkmark icon but eventually decided against it (as you can see in the final mockups).

waterim committed:
Merge remote-tracking branch 'upstream/main' into feature/download-option

waterim commented:
ESlint issue is related to [this](https://github.com/Expensify/App/issues/63254)

waterim committed:
Merge remote-tracking branch 'upstream/main' into feature/download-option

waterim committed:
fix conflict

flaviadefaria commented:
@waterim I was OoO yesterday, so only got to test the PR today.

This is what I see now when clicking the export option from the Inbox tab:

<img width="1495" alt="image" src="https://github.com/user-attachments/assets/55e36b73-ca54-4bc6-8a76-6e4df7b7934d" />

- I'm pretty sure that we only want to use the round logo, and from the latest mockups, it looks like we have settled for the grey icon. @Expensify/design can you please confirm that this is all correct? I see we had a bit of a discussion [here](https://github.com/Expensify/App/issues/58316#issuecomment-2723995443), but I want to confirm that this is where we landed.

- The integration options should be ordered as we have in Classic but with the new copy below:
    - Export to QuickBooks Online
    - Mark as exported
    - Basic export
    Any new template will come after these....

<img width="1491" alt="image" src="https://github.com/user-attachments/assets/38b7db28-3823-44e0-a20f-3e63e605c826" />


- I also noticed that in case the integration is broken that this is what the user sees:

<img width="1494" alt="image" src="https://github.com/user-attachments/assets/ef659795-5ac1-487c-bfad-e7c313ca71f0" />

In such a case, we should:
- Still order the export options as I mentioned above
-  Keep the integration icons instead of replacing them and updating `Export to QuickBooks Online` to `Export to Undefined`
- If the user selects `Export to QuickBooks Online` and the connections are broken, then I guess we could display an error modal? I also thought about routing them to the workspace to fix the connection, but the user exporting might not have access to do this, so probably the former is a better option. @trjExpensify, what do you think would be best here?

dannymcclain commented:
Ok here are those integration icons. I've exported them the same way we do for all our icons that show up in options rows.

[**integration icons 20x20.zip**](https://github.com/user-attachments/files/20712726/integration.icons.20x20.zip)

cc @Expensify/design I've [added these to Figma](https://www.figma.com/design/qQIIaxVUdtggrh8eg8bI5B/Brand-Guidelines---Product-UI?node-id=8172-23169&t=fI7gVe4BzDFg48mV-11) as well (but have not yet made them components yet)

waterim committed:
update order and icon

flaviadefaria commented:
@waterim regarding the broken connection, based on @trjExpensify's [comment](https://github.com/Expensify/App/pull/62340#issuecomment-2967702903) there should already be an in-line error on the system message when attempting to export. So for what concerns this change we just need to make sure that we continue to display the integration icons on both `Export to` and `Mark as exported` options even if the connection is broken.

waterim committed:
Merge remote-tracking branch 'upstream/main' into feature/download-option

waterim committed:
new icons

waterim committed:
new icon selector

waterim committed:
update icons

waterim commented:
@flaviadefaria Fixed everything except this: **`we just need to make sure that we continue to display the integration icons`**, How can I test failing integration, because for now Im not sure how can I do that as integrationConnection is coming as undefined and no way how to get a correct value, but need to test it to get an idea how can I do that

Also as I can see AI translations are here and how should I update translations for all the languages now? Didn't find any command for auto update

waterim committed:
add fallback for the broken integration

waterim commented:
<img width="352" alt="image" src="https://github.com/user-attachments/assets/acf85732-1cbc-404f-a866-ebc9d7c0b709" />

Updated, added a fallback when integration is not valid we will still show the icon from the connectedIntegration itself, even if its broken 
cc @flaviadefaria @trjExpensify

waterim committed:
update translations

waterim committed:
Merge remote-tracking branch 'upstream/main' into feature/download-option

waterim committed:
update translation

flaviadefaria commented:
@waterim any updates here?

waterim committed:
Merge remote-tracking branch 'upstream/main' into feature/download-option

waterim commented:
@flaviadefaria Missed that We've got the answer on the thread

> The fix there should have been for integrations that were never properly connected, which is not the case with the one that I'm testing. Here is a https://github.com/Expensify/App/issues/62451#issuecomment-2917600952 explaining the difference. So if that PR changed this for integrations that had already been connected but expired, then we should flag that in the issue.

Im still not sure what should I change as I didn't touch the behaviour for broken integrations and can't test it directly even with @trjExpensify account, can you please confirm that you can still see export options on the staging and can't see in my PR?

waterim commented:
The only thing I can see is that we have a check [here](https://github.com/Expensify/App/blob/a2bce5b827d18fc7ea4e2b6e5a1e5e14231a4e83/src/libs/ReportSecondaryActionUtils.ts#L535) and Im just reusing it for the push row, where we have this updated [valid integration](https://github.com/Expensify/App/blob/a2bce5b827d18fc7ea4e2b6e5a1e5e14231a4e83/src/libs/ReportSecondaryActionUtils.ts#L268) which I mentioned previously and which as I understand is changing behaviour, looks like a regression from [this ticket](https://github.com/Expensify/App/issues/62451)

flaviadefaria commented:
Still I don't see the options when the integration is broken in your PR:

<img width="1490" alt="image" src="https://github.com/user-attachments/assets/cc809733-5d80-4dd8-85ca-3123268a18e9" />

<img width="1490" alt="image" src="https://github.com/user-attachments/assets/0b941070-0204-4211-a749-228d7ce92e5a" />


<img width="1493" alt="image" src="https://github.com/user-attachments/assets/13828665-2a73-4306-bbc9-b73e9f35db60" />


And this is what I see in staging for the same report:

<img width="1494" alt="image" src="https://github.com/user-attachments/assets/470ac47e-0912-4e55-afa1-43ac6a960243" />

flaviadefaria commented:
So I guess the issue is that options are already not showing in staging when they should... I'll post in the other issue.

waterim commented:
@flaviadefaria All other issues should be fixed, right?
Is it a blocker for us, the connected integration?

flaviadefaria commented:
@waterim I just re-tested everything we're good to merge this. I'll push to resolve the broken connection issue in this [GH](https://github.com/Expensify/App/issues/62451) since that is where we changed the expected behavior. Thanks for all your help!

luacmartins commented:
@shubham1206agra let's prioritize reviewing this one please!

dannymcclain commented:
>Do we allow sub-menu highlighting here?

Yeah that looks like standard behavior for submenus to me. Here's another example from the product: adding an expense to a report.
![CleanShot 2025-06-25 at 09 00 55](https://github.com/user-attachments/assets/04cb617a-1e20-43a0-89d0-38c39c0cccd5)

shawnborton commented:
Yup, agree with that!

luacmartins commented on src/languages/pt-BR.ts:
```diff
@@ -546,6 +546,7 @@ const translations = {
         userID: 'ID do Usuário',
         disable: 'Desativar',
         export: 'Exportar',
+        basicExport: 'Exporta\u00E7\u00E3o b\u00E1sica',
```
NAB we can do this in a follow up

```suggestion
        basicExport: 'Exportação básica',
```

luacmartins reviewed (approved):
LGTM

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.
